frontend/components/MainPage/sections/AccountBalances/YourAgent.tsx

```diff
@@ -0,0 +1,177 @@
+
+import { Chain } from '@/client';
+import { AddressLink } from '@/components/AddressLink';
+import { SERVICE_REGISTRY_L2_CONTRACT_ADDRESS } from '@/constants/contractAddresses';
+import { UNICODE_SYMBOLS } from '@/constants/symbols';
+import { useAddress } from '@/hooks/useAddress';
+import { useBalance } from '@/hooks/useBalance';
+import { useReward } from '@/hooks/useReward';
+import { useServices } from '@/hooks/useServices';
+import { balanceFormat } from '@/utils/numberFormatters';
+import { truncateAddress } from '@/utils/truncate';
+
+import { InfoBreakdownList } from '../../../InfoBreakdown';
+import { Container, infoBreakdownParentStyle } from './styles';
+import {
+  OlasTitle,
+  OwnershipNftTitle,
+  ServiceIdTitle,
+  SignerTitle,
+  XdaiTitle,
+} from './Titles';
+
+const { Title } = Typography;
+
+const NftCard = styled(Card)`
+  .ant-card-body {
+    padding: 0;
+    img {
+      border-radius: 8px;
+    }
+  }
+`;
+
+const ServiceAndNftDetails = () => {
+  const { serviceId } = useServices();
+  const serviceAddress =
+    SERVICE_REGISTRY_L2_CONTRACT_ADDRESS[`${Chain.GNOSIS}`];
+
+  return (
+    <NftCard>
+      <Flex>
+        <Flex>
+          <Image width={78} height={78} alt="NFT" src="/NFT.png" />
+        </Flex>
+        <Flex
+          style={{ padding: '16px 12px' }}
+          align="center"
+          justify="space-between"
+          flex={1}
+        >
+          <Flex vertical>
+            <OwnershipNftTitle />
+            <a
+              href={`https://gnosis.blockscout.com/token/${serviceAddress}/instance/${serviceId}`}
+              target="_blank"
+            >
+              {truncateAddress(serviceAddress)} {UNICODE_SYMBOLS.EXTERNAL_LINK}
+            </a>
+          </Flex>
+
+          <Flex vertical>
+            <ServiceIdTitle />
+            <a
+              href={`https://registry.olas.network/gnosis/services/${serviceId}`}
+              target="_blank"
+            >
+              {serviceId} {UNICODE_SYMBOLS.EXTERNAL_LINK}
+            </a>
+          </Flex>
+        </Flex>
+      </Flex>
+    </NftCard>
+  );
+};
+
+export const YourAgentWallet = () => {
+  const { agentSafeBalance } = useBalance();
+  const { accruedServiceStakingRewards } = useReward();
+  const {
+    instanceAddress: agentInstanceAddress,
+    multisigAddress: agentSafeAddress,
+  } = useAddress();
+
+  const olasBalances = useMemo(() => {
+    return [
+      {
+        title: 'Claimed rewards',
+        value: balanceFormat(agentSafeBalance?.OLAS ?? 0, 2),
+      },
+      {
+        title: 'Unclaimed rewards',
+        value: balanceFormat(accruedServiceStakingRewards ?? 0, 2),
+      },
+    ];
+  }, [agentSafeBalance?.OLAS, accruedServiceStakingRewards]);
+
+  const xdaiBalances = useMemo(() => {
+    return [
+      {
+        title: <XdaiTitle />,
+        value: balanceFormat(agentSafeBalance?.ETH ?? 0, 2),
+      },
+    ];
+  }, [agentSafeBalance?.ETH]);
+
+  const signerInfo = useMemo(() => {
```

Comment:
same for this, maybe can just return 
```
[{
  left: <SignerTitle />,
  leftClassName: 'text-light text-sm',
  right: <AddressLink address={agentInstanceAddress} />,
  rightClassName: 'font-normal'
}]
```
since it exists only for the InfoBreakdownList